code/__HELPERS/plist.dm

```diff
@@ -0,0 +1,126 @@
+// PLISTS BY RPAUL
+// basically a better version of lists mimicing php lists
+// http://www.byond.com/docs/ref/info.html#/list/operators del this later
+// http://www.byond.com/docs/ref/info.html#/operator/overload this too
+
+plist
+    var/list/_list = list()
+
+/*
+=================
+    OPERATORS
+=================
+*/
+//return at idx TODO maybe sanitize. list["[var]" fix that somehow
+plist/proc/operator[](var/idx)
+    return _list[idx]
+
+//check if list needs to be extended, so if idx is num and > then len. if idx is str we dont care
+//if idx is null, append
+plist/proc/operator[]=(idx, B)
+    _list.Insert(idx, B)
+
+//actually add list as an element, not add all elements, also add everything else properly
+plist/proc/operator+(B)
+    src += B
+    return _list
```

Comment:
Thanks to aliasing this is practically `+=`

Reply:
oh shit thats wrong

Reply:
+ should only return the result, not modify the list